scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -704,6 +744,56 @@ impl TomlManifest {
     }
 }
 
+/// Returns the absolute canonical path of the README file for a [`TomlPackage`].
+pub fn readme_for_package(
+    package_root: &Utf8Path,
+    readme: Option<&PathOrBool>,
+) -> Result<Option<Utf8PathBuf>> {
+    match readme {
+        None => abs_canonical_path(
+            package_root,
+            default_readme_from_package_root(package_root.parent().unwrap()),
+        ),
+        Some(value) => match value {
+            PathOrBool::Bool(false) => Ok(None),
+            PathOrBool::Bool(true) => abs_canonical_path(package_root, Some("README.md".into())),
+            PathOrBool::Path(p) => abs_canonical_path(package_root, Some(p)),
+        },
+    }
+}
+
+/// Creates the absolute canonical path of the README file and checks if it exists
+fn abs_canonical_path(prefix: &Utf8Path, readme: Option<&Utf8Path>) -> Result<Option<Utf8PathBuf>> {
+    match readme {
+        None => Ok(None),
+        Some(readme) => prefix
+            .parent()
+            .unwrap()
+            .join(readme)
+            .canonicalize_utf8()
+            .map(Some)
+            .with_context(|| {
+                format!(
+                    "failed to find the readme at {}",
+                    prefix.parent().unwrap().join(readme)
+                )
+            }),
+    }
+}
+
+const DEFAULT_README_FILES: [&str; 3] = ["README.md", "README.txt", "README"];
+
+/// Checks if a file with any of the default README file names exists in the package root.
+/// If so, returns a `Utf8Path` with that name.
+fn default_readme_from_package_root(package_root: &Utf8Path) -> Option<&Utf8Path> {
+    for &readme_filename in DEFAULT_README_FILES.iter() {
```

Comment:
```suggestion
    for &readme_filename in DEFAULT_README_FILES {
```

there's builtin `IntoIter` implementation for `&[T]`